spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/mapping/SpannerPersistentEntityImpl.java

```diff
@@ -208,6 +209,12 @@ private void verifyEmbeddedColumnNameOverlap(Set<String> seen,
 		spannerPersistentEntity.doWithProperties(
 				(PropertyHandler<SpannerPersistentProperty>) spannerPersistentProperty -> {
 					if (spannerPersistentProperty.isEmbedded()) {
+						if (ConversionUtils.isIterableNonByteArrayType(
+								spannerPersistentProperty.getType())) {
+							throw new SpannerDataException(
+									"Embedded properties cannot be collections: "
```

Comment:
Is it a limitation of Spanner, or is it just to disambiguate with one-to-many relationship?

Reply:
I would say this is a way to disambiguate. There's nothing preventing a spanner parent row from having just 1 child row interleaved, but there is also no mechanism to enforce this 1 limit in Spanner.  As a result, I think there's much less opportunity for confusion and debugging work if we reflect that in our lib too: that it's inherently a list and there is no mechanism to enforce it to be 1 item always.